### DISCUSSION TITLE
Unable to handle Supabase errors in Flutter

### DISCUSSION THREAD

**Discussion by ThatLinuxGuyYouKnow** [supabase/supabase#13566] [Questions] [answered]
I'm trying to handle error that may arise while searching for users on my 'Users' table![Screenshot from 2023-04-10 07-17-27]
![Screenshot from 2023-04-10 07-21-18](https://user-images.githubusercontent.com/92615787/230840698-ed187a89-8fe4-4f76-a5d1-57b4dd21fda8.png)

![Screenshot from 2023-04-10 07-18-11](https://user-images.githubusercontent.com/92615787/230841671-86677b4f-e239-4d68-be86-591760c401d0.png)

However I'm unable to handle errors, because apparently 'error' is undefined

I'd appreciate any help

**Comment by GaryAustin1** [comment, marked as answer]
You have to use try and catch for errors with the flutter client.
See this guide for some examples... https://supabase.com/docs/guides/getting-started/tutorials/with-flutter
```
   try {
      final userId = supabase.auth.currentUser!.id;
      final data = await supabase
          .from('profiles')
          .select()
          .eq('id', userId)
          .single() as Map;
      _usernameController.text = (data['username'] ?? '') as String;
      _websiteController.text = (data['website'] ?? '') as String;
      _avatarUrl = (data['avatar_url'] ?? '') as String;
    } on PostgrestException catch (error) {
      context.showErrorSnackBar(message: error.message);
    } catch (error) {
      context.showErrorSnackBar(message: 'Unexpected exception occurred');
    }
```
